shell/src/main/java/org/jtaccuino/jshell/extensions/use/UseExtension.java

```diff
@@ -32,12 +39,67 @@ public Optional<String> shellVariableName() {
         return Optional.of("useExtension");
     }
 
+    private String toSnakeCaseEnumName(String extensionShortName) {
+        if (extensionShortName == null) {
+            return null;
+        }
+        if (extensionShortName.isEmpty()) {
+            return "";
+        }
+
+        StringBuilder snakeCaseBuilder = new StringBuilder();
+        int n = extensionShortName.length();
+
+        for (int i = 0; i < n; i++) {
+            char currentChar = extensionShortName.charAt(i);
+
+            // If it's an uppercase character
+            if (Character.isUpperCase(currentChar)) {
+                // Check if it's NOT the first character
+                // AND if the previous character was lowercase (e.g., "myVariable" -> "my_variable")
+                // OR if it's part of a sequence of uppercase characters where
+                //    the next character is lowercase (e.g., "HTTPRequest" -> "http_request")
+                if (i > 0 &&
+                    (Character.isLowerCase(extensionShortName.charAt(i - 1)) ||
+                     (i + 1 < n && Character.isLowerCase(extensionShortName.charAt(i + 1))))) {
+                    snakeCaseBuilder.append('_');
+                }
+                snakeCaseBuilder.append(Character.toLowerCase(currentChar));
+            } else {
+                // If it's a lowercase character, just append it
+                snakeCaseBuilder.append(currentChar);
+            }
+        }
+
+        return snakeCaseBuilder.toString().toUpperCase(Locale.ENGLISH);
+    }
+
     @Override
     public Optional<String> initCodeSnippet() {
-        return Optional.of("""
+        String enums = JShellExtension.Mode.ON_DEMAND.getExtensionClasses()
+                .stream()
+                .peek(System.out::println)
+                .map(e -> toSnakeCaseEnumName(e.getSimpleName()) + "(" + e.getName() + ".class)")
+                .collect(Collectors.joining(",\n"));
+
+        return Optional.of("enum JTExtension implements org.jtaccuino.jshell.extensions.use.UseExtension.OnDemandExtension {\n" + enums + ";\n"
+                + """
+                             final private Class<? extends org.jtaccuino.jshell.extensions.JShellExtension> extensionClass;
+
+                             public Class<? extends org.jtaccuino.jshell.extensions.JShellExtension> getExtensionClass() {
+                                return extensionClass;
+                             }
+
+                             JTExtension(Class<? extends org.jtaccuino.jshell.extensions.JShellExtension> exClass) {
+                                this.extensionClass = exClass;
+                             }
+                           }
```

Comment:
nitpick: irregular/uneven indentation
2 chars for class to method but
but method to method body has 2 chars outside the enum and 3 chars inside the enum